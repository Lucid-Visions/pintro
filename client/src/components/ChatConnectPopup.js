import React from "react";
import {
  Picker,
  TextInput,
  TouchableOpacity,
  View,
  StyleSheet,
  Text,
} from "react-native";
import Modal from "react-native-modal";
import TagsData from "../assets/TagsData";
import RNPickerSelect from "react-native-picker-select";


const pickerStyle = {
	inputAndroid: {
    width: 300,
	}
};

class ChatConnectPopup extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      intent: "",
      message: "",
      name: this.props.name,
      chat: this.props.chat, //"Chat" or "Connect"
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    this.props.toggle();
    this.props.onPress(this.state.intent, this.state.message);
    this.setState({
      intent: "",
      message: "",
    });
  }

  render() {
    const toggleModal = () => {
      this.props.toggle();
    };

    const intentListMap = TagsData.map((item) => ({
      label: item.text,
      value: item.text,
    }));
  
    const extraStyles = this.props.chat == "Connect" ? styles.connectPicker : styles.chatPicker

    let pickerStyles = {
      inputAndroid: {
        width: 300,
      },
      ...extraStyles
    }

    return (
      <Modal
        animationIn="slideInUp"
        animationOut="slideOutDown"
        isVisible={this.props.visible}
        onBackdropPress={() => toggleModal()}
        style={{ minHeight: 200 }}
      >
        <View
          style={[
            styles.chatContainer,
            this.props.chat == "Connect" && styles.connectContainer,
          ]}
        >
          <Text
            style={[
              styles.chatH1,
              this.props.chat == "Connect" && styles.connectH1,
            ]}
          >
            {this.props.chat} with {this.props.name}
          </Text>
          <View style={styles.content}>
            <Text
              style={
                this.props.chat == "Connect"
                  ? styles.connectLabel
                  : styles.chatLabel
              }
            >
              Select an intent
            </Text>

            <View style={{ marginLeft: 5 }}>
              <Text
                onPress={() => this.picker.togglePicker()}
                style={{
                  color: this.props.chat == "Connect" ? "white" : "black",
                  position: "absolute",
                  zIndex: 99,
                }}
              >
                {this.state.intent}
              </Text>
              <RNPickerSelect
                ref={(e) => (this.picker = e)}
                placeholder={{
                  label: "Select an option...",
                  value: null,
                }}
                selectedValue={this.state.intent}
                placeholderTextColor={this.props.chat == "Connect" ? "white" : "black"}
                style={pickerStyles}
                onValueChange={(itemValue) =>
                  this.setState({ intent: itemValue })
                }
                items={intentListMap}
              />
            </View>
            <Text
              style={{
                ...(this.props.chat == "Connect"
                  ? styles.connectLabel
                  : styles.chatLabel),
                marginTop: 20,
              }}
            >
              Write a message
            </Text>
            <TextInput
              style={[
                styles.chatText,
                this.props.chat == "Connect" && styles.connectText,
              ]}
              placeholderTextColor={
                this.props.chat == "Connect" ? "white" : "grey"
              }
              placeholder="Type here...    "
              onChangeText={(message) => this.setState({ message })}
              value={this.state.message}
            />
          </View>
          <TouchableOpacity
            style={[
              styles.chatButton,
              this.props.chat == "Connect" && styles.connectButton,
            ]}
            onPress={() => this.handleChange()}
          >
            <Text
              style={[
                styles.chatButtonText,
                this.props.chat == "Connect" && styles.connectButtonText,
              ]}
            >
              SEND {this.props.chat.toUpperCase()} REQUEST
            </Text>
          </TouchableOpacity>
        </View>
      </Modal>
    );
  }
}

const styles = StyleSheet.create({
  chatContainer: {
    flexDirection: "column",
    width: "95%",
    height: "50%",
    maxHeight: 550,
    backgroundColor: "white",
    borderRadius: 20,
    justifyContent: "space-between",
    alignSelf: "center",
    padding: "2%",
  },
  connectContainer: {
    flexDirection: "column",
    width: "95%",
    height: "50%",
    maxHeight: 550,
    backgroundColor: "black",
    borderRadius: 20,
    justifyContent: "space-between",
    alignSelf: "center",
    padding: "2%",
  },
  content: {
    alignItems: "flex-start",
    padding: "3%",
  },
  chatH1: {
    fontFamily: "poppins-bold",
    color: "black",
    margin: "auto",
    textAlign: "center",
    fontSize: 16,
    paddingTop: 10,
  },
  connectH1: {
    fontFamily: "poppins-bold",
    color: "white",
    margin: "auto",
    textAlign: "center",
    fontSize: 16,
    paddingTop: 10,
  },
  chatLabel: {
    fontFamily: "poppins-light",
    color: "#1A1A1A",
    margin: "auto",
    textAlign: "left",
    fontSize: 14,
    padding: "2%",
    paddingTop: "7%",
  },
  connectLabel: {
    fontFamily: "poppins-light",
    color: "grey",
    margin: "auto",
    textAlign: "left",
    fontSize: 14,
    padding: "2%",
    paddingTop: "7%",
  },
  chatText: {
    fontFamily: "poppins-light",
    color: "grey",
    margin: "auto",
    textAlign: "left",
    fontSize: 16,
    padding: "2%",
  },
  connectText: {
    fontFamily: "poppins-light",
    color: "white",
    margin: "auto",
    textAlign: "left",
    fontSize: 16,
    padding: "2%",
  },
  chatPicker: {
    width: "100%",
    fontFamily: "poppins-light",
    color: "grey",
  },
  connectPicker: {
    width: "100%",
    height: 40,
    fontFamily: "poppins-light",
    color: "white",
  },
  pickerItem: {
    fontFamily: "poppins-light",
    color: "grey",
  },
  chatButton: {
    flexDirection: "row",
    alignSelf: "center",
    alignItems: "center",
    justifyContent: "center",
    width: "90%",
    height: 30,
    backgroundColor: "black",
    borderRadius: 15 / 2,
  },
  chatButtonText: {
    marginLeft: "5%",
    color: "white",
    fontSize: 10,
  },
  connectButton: {
    flexDirection: "row",
    alignSelf: "center",
    alignItems: "center",
    justifyContent: "center",
    width: "90%",
    height: 30,
    backgroundColor: "white",
    borderRadius: 15 / 2,
  },
  connectButtonText: {
    marginLeft: "5%",
    color: "black",
    fontSize: 10,
  },
});

export default ChatConnectPopup;
