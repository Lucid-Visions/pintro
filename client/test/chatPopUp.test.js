import React from 'react';
import renderer, { fireEvent } from 'react-test-renderer';
import ChatConnectPopup from '../src/components/ChatConnectPopup';

let visible = true
const mockToggle = jest.fn(() => visible=false)

test('Chat pop up renders correctly', () => {
    const component = renderer.create(<ChatConnectPopup name="Bill Gates" chat="Chat" toggle={mockToggle}/>).toJSON();
    expect(component).toMatchSnapshot();
});

test('Chat pop up renders correctly with different params', () => {
    const component = renderer.create(<ChatConnectPopup name="Bill Gates" chat="Connect" toggle={mockToggle}/>).toJSON();
    expect(component).toMatchSnapshot();
});

test('test backdrop press', () => {
    const component = renderer.create(<ChatConnectPopup name="Bill Gates" chat="Connect" toggle={mockToggle}/>);
    const instance = component.root
    const modal = instance.findByType("Modal")
    const backdrop = modal.children[0].props

    expect(visible).toBe(true);
    // click on backdrop
    backdrop.onPress();
    expect(visible).toBe(false);
});

