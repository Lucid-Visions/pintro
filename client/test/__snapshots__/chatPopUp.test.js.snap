// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat pop up renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "space-between",
            "maxHeight": 550,
            "padding": "2%",
            "width": "95%",
          },
          false,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "poppins-bold",
              "fontSize": 16,
              "margin": "auto",
              "paddingTop": 10,
              "textAlign": "center",
            },
            false,
          ]
        }
      >
        Chat
         with 
        Bill Gates
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "padding": "3%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1A1A1A",
              "fontFamily": "poppins-light",
              "fontSize": 14,
              "margin": "auto",
              "padding": "2%",
              "paddingTop": "7%",
              "textAlign": "left",
            }
          }
        >
          Select an intent
        </Text>
        <View
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "black",
                "position": "absolute",
                "zIndex": 99,
              }
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ios_touchable_wrapper"
            >
              <View
                pointerEvents="box-only"
              >
                <TextInput
                  allowFontScaling={true}
                  editable={false}
                  rejectResponderTermination={true}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "black",
                      },
                    ]
                  }
                  testID="text_input"
                  underlineColorAndroid="transparent"
                  value="Select an option..."
                />
              </View>
            </View>
            <Modal
              hardwareAccelerated={false}
              onOrientationChange={[Function]}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              testID="ios_modal"
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
                testID="ios_modal_top"
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#f8f8f8",
                      "borderTopColor": "#dedede",
                      "borderTopWidth": 1,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "space-between",
                      "paddingHorizontal": 10,
                      "zIndex": 2,
                    },
                    undefined,
                  ]
                }
                testID="input_accessory_view"
              >
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "#a1a1a1",
                            "borderRightWidth": 1.5,
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "marginLeft": 11,
                            "transform": Array [
                              Object {
                                "translateY": 4,
                              },
                              Object {
                                "rotate": "-45deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "#a1a1a1",
                            "borderRightWidth": 1.5,
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "marginLeft": 22,
                            "transform": Array [
                              Object {
                                "translateY": -5,
                              },
                              Object {
                                "rotate": "135deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  hitSlop={
                    Object {
                      "bottom": 4,
                      "left": 4,
                      "right": 4,
                      "top": 4,
                    }
                  }
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="done_button"
                >
                  <View
                    testID="needed_for_touchable"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#007aff",
                            "fontSize": 17,
                            "fontWeight": "600",
                            "paddingRight": 11,
                            "paddingTop": 1,
                          },
                          undefined,
                          Object {},
                        ]
                      }
                      testID="done_text"
                    >
                      Done
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#d0d4da",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 215,
                    },
                    undefined,
                  ]
                }
              >
                <View>
                  <RCTPicker
                    items={
                      Array [
                        Object {
                          "label": "Select an option...",
                          "textColor": undefined,
                          "value": null,
                        },
                        Object {
                          "label": "Pitch Decks",
                          "textColor": undefined,
                          "value": "Pitch Decks",
                        },
                        Object {
                          "label": "Proposals",
                          "textColor": undefined,
                          "value": "Proposals",
                        },
                        Object {
                          "label": "User Experience",
                          "textColor": undefined,
                          "value": "User Experience",
                        },
                        Object {
                          "label": "Public Speaking",
                          "textColor": undefined,
                          "value": "Public Speaking",
                        },
                        Object {
                          "label": "Apps",
                          "textColor": undefined,
                          "value": "Apps",
                        },
                        Object {
                          "label": "Graphic Design",
                          "textColor": undefined,
                          "value": "Graphic Design",
                        },
                        Object {
                          "label": "Technology",
                          "textColor": undefined,
                          "value": "Technology",
                        },
                        Object {
                          "label": "Websites",
                          "textColor": undefined,
                          "value": "Websites",
                        },
                        Object {
                          "label": "Marketing",
                          "textColor": undefined,
                          "value": "Marketing",
                        },
                        Object {
                          "label": "Social Media",
                          "textColor": undefined,
                          "value": "Social Media",
                        },
                        Object {
                          "label": "SEO",
                          "textColor": undefined,
                          "value": "SEO",
                        },
                        Object {
                          "label": "Wireframing",
                          "textColor": undefined,
                          "value": "Wireframing",
                        },
                      ]
                    }
                    onChange={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    selectedIndex={0}
                    style={
                      Array [
                        Object {
                          "height": 216,
                        },
                        undefined,
                      ]
                    }
                    testID="ios_picker"
                  />
                </View>
              </View>
            </Modal>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#1A1A1A",
              "fontFamily": "poppins-light",
              "fontSize": 14,
              "margin": "auto",
              "marginTop": 20,
              "padding": "2%",
              "paddingTop": "7%",
              "textAlign": "left",
            }
          }
        >
          Write a message
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Type here...    "
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "grey",
                "fontFamily": "poppins-light",
                "fontSize": 16,
                "margin": "auto",
                "padding": "2%",
                "textAlign": "left",
              },
              false,
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 7.5,
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 10,
                "marginLeft": "5%",
              },
              false,
            ]
          }
        >
          SEND 
          CHAT
           REQUEST
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`Chat pop up renders correctly with different params 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "space-between",
            "maxHeight": 550,
            "padding": "2%",
            "width": "95%",
          },
          Object {
            "alignSelf": "center",
            "backgroundColor": "black",
            "borderRadius": 20,
            "flexDirection": "column",
            "height": "50%",
            "justifyContent": "space-between",
            "maxHeight": 550,
            "padding": "2%",
            "width": "95%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontFamily": "poppins-bold",
              "fontSize": 16,
              "margin": "auto",
              "paddingTop": 10,
              "textAlign": "center",
            },
            Object {
              "color": "white",
              "fontFamily": "poppins-bold",
              "fontSize": 16,
              "margin": "auto",
              "paddingTop": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        Connect
         with 
        Bill Gates
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "padding": "3%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "grey",
              "fontFamily": "poppins-light",
              "fontSize": 14,
              "margin": "auto",
              "padding": "2%",
              "paddingTop": "7%",
              "textAlign": "left",
            }
          }
        >
          Select an intent
        </Text>
        <View
          style={
            Object {
              "marginLeft": 5,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "white",
                "position": "absolute",
                "zIndex": 99,
              }
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="ios_touchable_wrapper"
            >
              <View
                pointerEvents="box-only"
              >
                <TextInput
                  allowFontScaling={true}
                  editable={false}
                  rejectResponderTermination={true}
                  style={
                    Array [
                      undefined,
                      Object {
                        "color": "white",
                      },
                    ]
                  }
                  testID="text_input"
                  underlineColorAndroid="transparent"
                  value="Select an option..."
                />
              </View>
            </View>
            <Modal
              hardwareAccelerated={false}
              onOrientationChange={[Function]}
              supportedOrientations={
                Array [
                  "portrait",
                  "landscape",
                ]
              }
              testID="ios_modal"
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "opacity": 1,
                  }
                }
                testID="ios_modal_top"
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#f8f8f8",
                      "borderTopColor": "#dedede",
                      "borderTopWidth": 1,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "space-between",
                      "paddingHorizontal": 10,
                      "zIndex": 2,
                    },
                    undefined,
                  ]
                }
                testID="input_accessory_view"
              >
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "#a1a1a1",
                            "borderRightWidth": 1.5,
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "marginLeft": 11,
                            "transform": Array [
                              Object {
                                "translateY": 4,
                              },
                              Object {
                                "rotate": "-45deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    isTVSelectable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "#a1a1a1",
                            "borderRightWidth": 1.5,
                            "borderTopWidth": 1.5,
                            "height": 15,
                            "width": 15,
                          },
                          undefined,
                          Object {
                            "marginLeft": 22,
                            "transform": Array [
                              Object {
                                "translateY": -5,
                              },
                              Object {
                                "rotate": "135deg",
                              },
                            ],
                          },
                          undefined,
                          Object {},
                        ]
                      }
                    />
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  hitSlop={
                    Object {
                      "bottom": 4,
                      "left": 4,
                      "right": 4,
                      "top": 4,
                    }
                  }
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="done_button"
                >
                  <View
                    testID="needed_for_touchable"
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#007aff",
                            "fontSize": 17,
                            "fontWeight": "600",
                            "paddingRight": 11,
                            "paddingTop": 1,
                          },
                          undefined,
                          Object {},
                        ]
                      }
                      testID="done_text"
                    >
                      Done
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#d0d4da",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 215,
                    },
                    undefined,
                  ]
                }
              >
                <View>
                  <RCTPicker
                    items={
                      Array [
                        Object {
                          "label": "Select an option...",
                          "textColor": undefined,
                          "value": null,
                        },
                        Object {
                          "label": "Pitch Decks",
                          "textColor": undefined,
                          "value": "Pitch Decks",
                        },
                        Object {
                          "label": "Proposals",
                          "textColor": undefined,
                          "value": "Proposals",
                        },
                        Object {
                          "label": "User Experience",
                          "textColor": undefined,
                          "value": "User Experience",
                        },
                        Object {
                          "label": "Public Speaking",
                          "textColor": undefined,
                          "value": "Public Speaking",
                        },
                        Object {
                          "label": "Apps",
                          "textColor": undefined,
                          "value": "Apps",
                        },
                        Object {
                          "label": "Graphic Design",
                          "textColor": undefined,
                          "value": "Graphic Design",
                        },
                        Object {
                          "label": "Technology",
                          "textColor": undefined,
                          "value": "Technology",
                        },
                        Object {
                          "label": "Websites",
                          "textColor": undefined,
                          "value": "Websites",
                        },
                        Object {
                          "label": "Marketing",
                          "textColor": undefined,
                          "value": "Marketing",
                        },
                        Object {
                          "label": "Social Media",
                          "textColor": undefined,
                          "value": "Social Media",
                        },
                        Object {
                          "label": "SEO",
                          "textColor": undefined,
                          "value": "SEO",
                        },
                        Object {
                          "label": "Wireframing",
                          "textColor": undefined,
                          "value": "Wireframing",
                        },
                      ]
                    }
                    onChange={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    selectedIndex={0}
                    style={
                      Array [
                        Object {
                          "height": 216,
                        },
                        undefined,
                      ]
                    }
                    testID="ios_picker"
                  />
                </View>
              </View>
            </Modal>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "grey",
              "fontFamily": "poppins-light",
              "fontSize": 14,
              "margin": "auto",
              "marginTop": 20,
              "padding": "2%",
              "paddingTop": "7%",
              "textAlign": "left",
            }
          }
        >
          Write a message
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Type here...    "
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "color": "grey",
                "fontFamily": "poppins-light",
                "fontSize": 16,
                "margin": "auto",
                "padding": "2%",
                "textAlign": "left",
              },
              Object {
                "color": "white",
                "fontFamily": "poppins-light",
                "fontSize": 16,
                "margin": "auto",
                "padding": "2%",
                "textAlign": "left",
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderRadius": 7.5,
            "flexDirection": "row",
            "height": 30,
            "justifyContent": "center",
            "opacity": 1,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 10,
                "marginLeft": "5%",
              },
              Object {
                "color": "black",
                "fontSize": 10,
                "marginLeft": "5%",
              },
            ]
          }
        >
          SEND 
          CONNECT
           REQUEST
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
